### ISSUE TITLE
Missing api-ms-win-core-winrt-l1-1-0.dll

### ISSUE THREAD

**Issue by wf-soft** [dotnet/core#6976]
Many win7 systems are prompted that api-ms-win-core-winrt-l1-1-0.dll is missing. I use. Net 6.0 independent application to publish. The current solution is to download api-ms-win-core-winrt-l1-1-0.dll and put it into C: Windows\system32 Is there a better solution

**Comment by mairaw** [comment]
@leecow @richlander can someone comment here on the dependencies needed to install .NET apps on Windows 7 machines?

**Comment by sencagri** [comment]
Yep! Same problem here. Doesself contained publishing does not include these libraries? Am I doing something wrong?

**Comment by jeffhandley** [comment]
Hello @wf-soft and @sencagri. Can you confirm you have the prerequisites installed that are listed here please?
https://docs.microsoft.com/en-us/dotnet/core/install/windows?tabs=net60#additional-deps

More information about Windows 7 ESU can be found here:
https://docs.microsoft.com/en-us/troubleshoot/windows-client/windows-7-eos-faq/windows-7-extended-security-updates-faq

**Comment by wf-soft** [comment]
@jeffhandley I'm not sure, because these problems occur on some users' machines. At present, api-ms-win-core-winrt-l1-1-0.dll is added to the program directory  solves this problem

**Comment by joostvz** [comment]
@wf-soft FWIW: I was able to fix this error by building my .NET 6 app with the following runtime identifier: "win7-x64". It doesn't add the api-ms-win-core-winrt-l1-1-0.dll to the output folder, but it does add several other api-ms-win-core-xxx-l1-1-0 dll's.

**Comment by jeffhandley** [comment]
Thanks for sharing that solution, @joostvz. @wf-soft, can you test that solution out for your scenario too?

**Comment by sinfere** [comment]
I meet the same problem, while I can resolve it by uncheck the option "Produce single file"
But any solutions when option "Produce single file" checked?

**Comment by sinfere** [comment]
![image](https://user-images.githubusercontent.com/1476105/152634348-894f725d-262a-40b9-a9b0-702eb6d6c473.png)

**Comment by thepwrtank18** [comment]
It looks like api-ms-win-core-winrt-l1-1-0.dll is missing from .NET installations (C:\Program Files\dotnet\shared\Microsoft.NETCore.App\6.0.9).

**Comment by thepwrtank18** [comment]
> @wf-soft FWIW: I was able to fix this error by building my .NET 6 app with the following runtime identifier: "win7-x64". It doesn't add the api-ms-win-core-winrt-l1-1-0.dll to the output folder, but it does add several other api-ms-win-core-xxx-l1-1-0 dll's.

This still shows the core-winrt error.
